src/jsrm/systems/planar_pcs_num.py

```diff
@@ -0,0 +1,1280 @@
+            Returns:
+                S: elastic matrix of shape (n_q, n_q) if formulate_in_strain_space is False or (n_xi, n_xi) otherwise
+            """
+            # length of the segments
+            l = params["l"]
+            # cross-sectional area and second moment of area
+            A = jnp.pi * params["r"] ** 2
+            Ib = A**2 / (4 * jnp.pi)
+
+            # elastic and shear modulus
+            E, G = params["E"], params["G"]
+            # stiffness matrix of shape (num_segments, 3, 3)
+            S_sms = compute_stiffness_matrix_for_all_segments_fn(l, A, Ib, E, G)
+            # we define the elastic matrix of shape (n_xi, n_xi) as K(xi) = K @ xi where K is equal to
+            S = blk_diag(S_sms)
+
+            if not formulate_in_strain_space:
+                S = B_xi.T @ S @ B_xi
+
+            return S
+    if not isinstance(stiffness_fn, Callable):
+        raise TypeError(f"stiffness_fn must be a callable, but got {type(stiffness_fn).__name__}")
+
+    # Actuation mapping function
+    if actuation_mapping_fn is None: 
+        def actuation_mapping_fn(
+            forward_kinematics_fn: Callable,
+            jacobian_fn: Callable,
+            params: Dict[str, Array],
+            B_xi: Array,
+            q: Array,
+        ) -> Array:
+            """
+            Returns the actuation matrix that maps the actuation space to the configuration space.
+            Assumes the fully actuated and identity actuation matrix.
+            Args:
+                forward_kinematics_fn: function to compute the forward kinematics
+                jacobian_fn: function to compute the Jacobian
+                params: dictionary with robot parameters
+                B_xi: strain basis matrix
+                q: configuration of the robot
+            Returns:
+                A: actuation matrix of shape (n_xi, n_xi) where n_xi is the number of strains.
+            """
+            A = B_xi.T @ jnp.identity(n_xi) @ B_xi
+
+            return A
+    if not isinstance(actuation_mapping_fn, Callable):
+        raise TypeError(f"actuation_mapping_fn must be a callable, but got {type(actuation_mapping_fn).__name__}")
+    
+    if integration_type == "gauss":
+        if param_integration is None:
+            param_integration = 30
+    elif integration_type == "trapezoid":
+        if param_integration is None:
+            param_integration = 1000
+    else:
+        raise ValueError(f"integration_type must be either 'gauss' or 'trapezoid', but got {integration_type}")
+    
+    # =======================================================================================================================
+    # Define the functions
+    # ====================================================
+    
+    # Compute the strain basis matrix
+    B_xi = compute_strain_basis(strain_selector)
+    
+    @jit
+    def apply_eps_to_bend_strains(xi: Array, _eps: float) -> Array:
```

Comment:
Please remove this as I think the new explicit implementation doesn't exhibit the singularities for `\kappa_be = 0` anymore, right?

Reply:
I see! Actually, let's have a more in-depth discussion about this what would be the best way of handling it in order to obtain a good differentiability. 

How does SoRoSim tackle this issue? With a separate expression for the adjoint and tangent operators?

Reply:
It might also make sense to check the broader literature if there has been solutions proposed to obtain differentiability for the adjoint and tangent operators, even outside the soft robotics community.

Reply:
Yes, if you apply a similar strategy that avoid singularities based on a condition + expansion of the terms for the forward kinematics (i.e., `chi_fn`), then this should work without the `apply_eps_bending_strains`, right?